src/citrine/_rest/collection.py

```diff
@@ -98,6 +104,70 @@ def list(self,
                 # properly, so we are filtering client-side.
                 pass
 
+    def list(self,
+             page: Optional[int] = None,
+             per_page: Optional[int] = None) -> Iterable[ResourceType]:
+        """
+        List all visible elements in the collection.
+
+        Leaving page and per_page as default values will yield all elements in the
+        collection, paginating over all available pages.
+
+        Parameters
+        ---------
+        page: int, optional
+            The "page" of results to list. Default is to read all pages and yield
```

Comment:
FWIW I would support deprecating the `page` parameter now that you have this default behavior.

Reply:
That's a good point.  I don't want to break any existing clients so raising a deprecation warning is a good idea.